src/dsp_tools/commands/validate_data/validate_data.py

```diff
@@ -75,15 +77,17 @@ def validate_data(filepath: Path, api_url: str, save_graphs: bool) -> bool:
         print(VALIDATION_ERRORS_FOUND_MSG)
         print(msg)
         # if the ontology itself has errors, we will not validate the data
-        return True
+        return False
 
     report = _get_validation_result(graphs, shacl_validator, save_path)
     if report.conforms:
         logger.info("Validation passed.")
         print(BACKGROUND_BOLD_GREEN + "\n   Validation passed!   " + RESET_TO_DEFAULT)
-    else:
-        _print_shacl_validation_violation_message(report, filepath, save_graphs)
-    return True
+        return True
+    reformatted = reformat_validation_graph(report)
+    sorted_problems = sort_user_problems(reformatted)
+    _print_shacl_validation_violation_message(sorted_problems, report, filepath, save_graphs)
+    return any([bool(sorted_problems.unique_violations), bool(sorted_problems.unexpected_shacl_validation_components)])
```

Comment:
Do I understand correctly that now it returns True if errors were found? I would return False then...

Plus, I think now the docstring is outdated. It says: 

```
Returns:
        true unless it crashed
```